src/cmd/cli/command/compose.go

```diff
@@ -514,6 +514,85 @@ func makeComposeLogsCmd() *cobra.Command {
 	return logsCmd
 }
 
+func makeComposeEstimateCmd() *cobra.Command {
+	var estimateCmd = &cobra.Command{
+		Use:         "estimate",
+		Args:        cobra.NoArgs,
+		Annotations: authNeededAnnotation,
+		Short:       "Estimate the cost of deploying the current project",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			region, _ := cmd.Flags().GetString("region")
+			os.Setenv("DEFANG_JSON", "1") // always show JSON output for estimate
+			loader := configureLoader(cmd)
+			project, err := loader.LoadProject(cmd.Context())
+			if err != nil {
+				return err
+			}
+
+			provider, err := getProvider(cmd.Context(), loader)
+			if err != nil {
+				return err
+			}
+
+			err = canIUseProvider(ctx, provider, project.Name)
+			if err != nil {
+				return err
+			}
+
+			since := time.Now()
+
+			resp, project, err := cli.ComposeUp(ctx, project, client, provider, compose.UploadModePreview, mode.Value())
+			if err != nil {
+				return err
+			}
+
+			var pulumiPreviewLogLines []string
+			options := cli.TailOptions{
+				Deployment: resp.Etag,
+				Since:      since,
+				LogType:    logs.LogTypeBuild,
+				Verbose:    true,
+			}
+
+			err = cli.TailAndWaitForCD(ctx, project.Name, provider, options, func(entry *defangv1.LogEntry, options *cli.TailOptions) error {
+				pulumiPreviewLogLines = append(pulumiPreviewLogLines, entry.GetMessage())
+				return nil
+			})
+			if err != nil {
+				return fmt.Errorf("failed to tail and wait for cd: %w", err)
```

Comment:
Errors from here are user facing. Do we really need the prefix?

Reply:
I think we should always prefix errors. It helps track down where and why an error happened. If you don't like the prefix, we can change it though.